src/network/codec.ts

```diff
@@ -0,0 +1,147 @@
+  let encodedBody: Buffer;
+  switch (method) {
+    case Method.Hello:
+      encodedBody = serialize(body, Hello);
+      break;
+    case Method.Goodbye:
+      encodedBody = serialize(body, Goodbye);
+      break;
+    case Method.Status:
+      encodedBody = serialize(body, Status);
+      break;
+    case Method.BeaconBlockRoots:
+      encodedBody = serialize(body, BeaconBlockRootsRequest);
+      break;
+    case Method.BeaconBlockHeaders:
+      encodedBody = serialize(body, BeaconBlockHeadersRequest);
+      break;
+    case Method.BeaconBlockBodies:
+      encodedBody = serialize(body, BeaconBlockBodiesRequest);
+      break;
+    case Method.BeaconStates:
+      encodedBody = serialize(body, BeaconStatesRequest);
+      break;
+    default:
+      throw new Error(`Invalid method ${method}`);
+  }
+  const request: WireRequest = {
+    id: Buffer.from(id, 'hex'),
+    method: method,
+    body: encodedBody,
+  };
+  return serialize(request, WireRequest);
+}
+
+export function encodeResponse(id: RequestId, method: Method, responseCode: number, result: ResponseBody): Buffer {
+  const response: WireResponse = {
+    id: Buffer.from(id, 'hex'),
+    responseCode,
+    result: Buffer.alloc(0),
+  };
+  if (responseCode !== 0) { // error response
+    return serialize(response, WireResponse);
+  }
+  switch (method) {
+    case Method.Hello:
+      response.result = serialize(result, Hello);
+      break;
+    case Method.Goodbye:
+      response.result = serialize(result, Goodbye);
+      break;
+    case Method.Status:
+      response.result = serialize(result, Status);
+      break;
+    case Method.BeaconBlockRoots:
+      response.result = serialize(result, BeaconBlockRootsResponse);
+      break;
+    case Method.BeaconBlockHeaders:
+      response.result = serialize(result, BeaconBlockHeadersResponse);
+      break;
+    case Method.BeaconBlockBodies:
+      response.result = serialize(result, BeaconBlockBodiesResponse);
+      break;
+    case Method.BeaconStates:
+      response.result = serialize(result, BeaconStatesResponse);
+      break;
+    default:
+      throw new Error(`Invalid method ${method}`);
+  }
+  return serialize(response, WireResponse);
+}
+
+// Decode
+
+export function decodeRequestBody(method: Method, body: Buffer): RequestBody {
+  switch (method) {
+    case Method.Hello:
+      return deserialize(body, Hello);
+    case Method.Goodbye:
+      return deserialize(body, Goodbye);
+    case Method.Status:
+      return deserialize(body, Status);
+    case Method.BeaconBlockRoots:
+      return deserialize(body, BeaconBlockRootsRequest);
+    case Method.BeaconBlockHeaders:
+      return deserialize(body, BeaconBlockHeadersRequest);
+    case Method.BeaconBlockBodies:
+      return deserialize(body, BeaconBlockBodiesRequest);
+    case Method.BeaconStates:
+      return deserialize(body, BeaconStatesRequest);
+    default:
+      throw new Error(`Invalid method ${method}`);
```

Comment:
Probably want to do something like drop the peer instead? Could propagate an error out?

Reply:
I think we do want a throw, it should just be caught by the calling code. I'll fix that.
The thing is we cannot create a valid return type if the `Method` is invalid.

Reply:
Where is the calling code? I can imagine a scenario where lets say an outdated Lodestar client accidentally DOSs us because this would throw. Inversely if this exsits a malicious actor could easily modify the `encodeResponse` or encodeRequest` with the intention to error our all the lodestar peers

Reply:
The calling code is in the rpc controller, in the onRequestResponse method. I don't understand the DOS.
If a peer sends us garbage, we throw it away. We `try` to `decodeRequest` or `decodeResponse` and if we `catch` an error, we bail and ignore the message.
In a future more advanced version, we can keep track of peers that send bad messages repeatedly and remove them.